Tasks/JenkinsQueueJobV2/jenkinsqueuejobtask.ts

```diff
@@ -127,18 +132,24 @@ async function doWork() {
         tl.setVariable('JENKINS_JOB_ID', rootJob.ExecutableNumber.toString());
     } catch (e) {
         let message: string;
-        if (e instanceof util.HttpError) {
-            message = e.message;
-            console.error(e.fullMessage);
-            console.error(e.body);
-        } else if (e instanceof Error) {
-            message = e.message;
-            console.error(e);
+        if (e.fullMessage.includes('HttpResponse.statusCode=302') && taskOptions.considerCode302AsSuccess) {
```

Comment:
Is it reliable to check presence of 'HttpResponse.statusCode=302'? What if it will be presented in error message for some other message not related to original issue here?
Have you checked e.body? If there is some json response - could it make sense to parse it instread and extract http return code?

Reply:
I didn't check e.body because I don't have repro of the issue. I have only some info provided by customer in ticket https://github.com/microsoft/azure-pipelines-tasks/issues/15455. Logs of affected job contain following lines:
```
Job creation failed.
HttpResponse.statusCode=302
HttpResponse.statusMessage=Found
```

Line `Job creation failed.` is provided to logs by one of 2 lines in code: [1](https://github.com/denis-tikhomirov/azure-pipelines-tasks/blob/d26c5ee1c01ec123c1f43f6be9eee83a50009a29/Tasks/JenkinsQueueJobV2/util.ts#L273) and [2](https://github.com/denis-tikhomirov/azure-pipelines-tasks/blob/d26c5ee1c01ec123c1f43f6be9eee83a50009a29/Tasks/JenkinsQueueJobV2/util.ts#L280). Next 2 lines are placed in e.fullMessage. fullMessage is created [here](https://github.com/denis-tikhomirov/azure-pipelines-tasks/blob/d26c5ee1c01ec123c1f43f6be9eee83a50009a29/Tasks/JenkinsQueueJobV2/util.ts#L16).
So I have ability to check only 2 values provided for http responce: httpResponse.statusCode and httpResponse.statusMessage. Check for 'statusCode = 302' is already added to IF statement. And I can add check for statusMessage, but each status code has only 1 status message ('302 - Found' in this specific case). And it's unnecessary to add check for status message.

Reply:
Why can't we extend HttpError class and pass http code - to be able to check it further?
https://github.com/denis-tikhomirov/azure-pipelines-tasks/blob/d26c5ee1c01ec123c1f43f6be9eee83a50009a29/Tasks/JenkinsQueueJobV2/util.ts#L45

Reply:
Really, I thought that it's complication of code the sake of only 1 case. But I have added this to my changes as you think it's better solution.

Reply:
Looks good, thanks! The main issue here is that searching of specific test like 'statusCode = 302' does not seem to be reliable solution - it could present here for any other reason. Or if something will be changed in error interface - this logic will be broken (with current changes interface changes will be caught by compiler).